ui/app/src/components/SingleItemSelector/SingleItemSelector.tsx

```diff
@@ -435,7 +439,9 @@ export function SingleItemSelector(props: SingleItemSelectorProps) {
       >
         <Breadcrumbs
           keyword={state?.keywords}
-          breadcrumb={state.breadcrumb.map((path) => state.byId[path] ?? state.byId[withIndex(path)])}
+          breadcrumb={state.breadcrumb
+            .map((path) => state.byId[path] ?? state.byId[withIndex(path)])
+            .filter((i) => i !== undefined)}
```

Comment:
Use flatMap to map and filter at once. 
```suggestion
          breadcrumb={state.breadcrumb
            .flatMap((path) => state.byId[path] ?? state.byId[withIndex(path)] ?? [])}
```